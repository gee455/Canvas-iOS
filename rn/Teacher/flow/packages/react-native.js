//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
