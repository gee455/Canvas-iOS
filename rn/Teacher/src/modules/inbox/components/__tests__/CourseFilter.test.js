//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* eslint-disable flowtype/require-valid-file-annotation */
import 'react-native'
import React from 'react'
import CourseFilter from '../CourseFilter'
import explore from '../../../../../test/helpers/explore'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const template = {
  ...require('../../../../__templates__/course'),
}

jest.mock('TouchableOpacity', () => 'TouchableOpacity')
jest.mock('../../../../routing')
const mockActionSheet = jest.fn()
jest.mock('ActionSheetIOS', () => ({
  showActionSheetWithOptions: mockActionSheet,
}))

let defaultCourses = [template.course(), template.course()]
let defaultProps = {
  courses: defaultCourses,
  selectedCourse: 'all',
  onClearFilter: jest.fn(),
  onSelectFilter: jest.fn(),
}

describe('CourseFilter', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <CourseFilter { ...defaultProps } />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the ActionSheet', () => {
    const tree = renderer.create(
      <CourseFilter { ...defaultProps } />
    ).toJSON()
    const button = explore(tree).selectByID('inbox.filterByCourse') || {}
    button.props.onPress()
    expect(mockActionSheet).toHaveBeenCalled()
  })

  it('calls the clearFilterCallback', () => {
    let props = {
      ...defaultProps,
      selectedCourse: 0,
    }
    const tree = renderer.create(
      <CourseFilter { ...props } />
    ).toJSON()
    const button = explore(tree).selectByID('inbox.filterByCourse') || {}
    button.props.onPress()
    expect(props.onClearFilter).toHaveBeenCalled()
  })

  it('calls the updateFilterCallback', () => {
    const tree = renderer.create(
      <CourseFilter { ...defaultProps } />
    )
    const instance = tree.getInstance()
    instance.updateFilter(1)
    expect(defaultProps.onSelectFilter).toHaveBeenCalled()
  })
})
