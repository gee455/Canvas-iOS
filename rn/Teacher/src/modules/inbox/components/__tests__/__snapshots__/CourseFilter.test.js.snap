// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseFilter renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      accessibilityTraits="header"
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontFamily": ".SFUIDisplay-semibold",
            "fontSize": 20,
          },
          Object {
            "color": "#2d3b44",
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "600",
          },
        ]
      }
    >
      All Courses
    </Text>
    <TouchableOpacity
      accessibilityLabel="Filter Inbox"
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      style={
        Object {
          "flexShrink": 0,
          "marginBottom": 1,
          "marginLeft": 8,
        }
      }
      testID="inbox.filterByCourse"
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 14,
                },
                Object {
                  "fontFamily": ".SFUIDisplay-bold",
                },
                Object {
                  "color": "#374A59",
                },
                Object {},
              ],
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Filter
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
